source/plugins/languages/visual/libraries/micropython/visual/definitions_micropython.js

```diff
@@ -0,0 +1,616 @@
+// DO NOT EDIT THIS FILE, IT IS AUTMATICALLY GENERATED
+
+module.exports = function (blockly){
+	var Blockly = blockly.Blockly;
+
+	Blockly.Blocks['micropython_pin'] = {
+		init: function() {
+			this.appendValueInput('GPIO')
+				.setCheck('Number')
+				.appendField('Pin');
+			this.appendDummyInput()
+				.appendField(new Blockly.FieldDropdown([['IN','IN'], ['OUT','OUT'], ['PULL_UP','PULL_UP']]), 'type1');
+			this.appendDummyInput()
+				.appendField(new Blockly.FieldDropdown([['none','NONE'], ['IN','IN'], ['OUT','OUT'], ['PULL_UP','PULL_UP']]), 'type2');
+			this.setInputsInline(true);
+			this.setOutput(true, null);
+			this.setColour(255);
+			this.setTooltip('');
+			this.setHelpUrl('');
+		}
+	};
+
+	Blockly.Blocks['micropython_set_value'] = {
+		init: function() {
+			this.appendValueInput('pin_setValue')
+				.setCheck('Variables')
+				.appendField('set value')
+				.appendField(new Blockly.FieldDropdown([['HIGH','0'], ['LOW','1']]), 'set_value')
+				.appendField('for');
+			this.setPreviousStatement(true, null);
+			this.setNextStatement(true, null);
+			this.setColour(230);
+			this.setTooltip('');
+			this.setHelpUrl('');
+		}
+	};
+
+	Blockly.Blocks['micropython_get_value'] = {
+		init: function() {
+			this.appendValueInput('get_value')
+				.setCheck('Variables')
+				.appendField('get value of');
+			this.setOutput(true, null);
+			this.setColour(230);
+			this.setTooltip('');
+			this.setHelpUrl('');
+		}
+	};
+
+	Blockly.Blocks['micropython_on_off'] = {
+		init: function() {
+			this.appendValueInput('set_valueLED')
+				.setCheck('Variables')
+				.appendField('set LED')
+				.appendField(new Blockly.FieldDropdown([['on','on'], ['off','off']]), 'set_on/off');
+			this.setPreviousStatement(true, null);
+			this.setNextStatement(true, null);
+			this.setColour(230);
+			this.setTooltip('');
+			this.setHelpUrl('');
+		}
+	};
+
+	Blockly.Blocks['micropython_pin_number'] = {
+		init: function() {
+			this.appendValueInput('pin_number')
+				.setCheck('Number')
+				.appendField('Pin');
+			this.setOutput(true, null);
+			this.setColour(255);
+			this.setTooltip('');
+			this.setHelpUrl('');
+		}
+	};
```

Comment:
I suggest not having this block, as there are many boards with micropython and we have no idea how many pins they have. Instead of using this block, just use a normal number block.